automation/taskcluster/decision_task.py

```diff
@@ -42,65 +43,67 @@
 )
 
 
-def pr_or_push():
+def pr_or_push(queue):
     if SKIP_TASKS_TRIGGER in PR_TITLE:
         print("Pull request title contains", SKIP_TASKS_TRIGGER)
         print("Exit")
         return {}
 
-    print("Fetching build variants from gradle")
     variants = build_variants.from_gradle()
 
-    if len(variants) == 0:
-        raise ValueError("Could not get build variants from gradle")
-
-    print("Got variants: {}".format(' '.join(variants)))
+    task_graph = TaskGraph()
+    for variant in variants:
+        task_graph.add_task(BUILDER.craft_assemble_task(variant))
+        task_graph.add_task(BUILDER.craft_test_task(variant))
 
-    build_tasks = {}
-    other_tasks = {}
+    task_graph.add_task(BUILDER.craft_detekt_task())
+    task_graph.add_task(BUILDER.craft_ktlint_task())
+    task_graph.add_task(BUILDER.craft_lint_task())
+    task_graph.add_task(BUILDER.craft_compare_locales_task())
 
-    for variant in variants:
-        build_tasks[taskcluster.slugId()] = BUILDER.craft_assemble_task(variant)
-        build_tasks[taskcluster.slugId()] = BUILDER.craft_test_task(variant)
+    return task_graph.schedule(queue)
 
-    for craft_function in (
-        BUILDER.craft_detekt_task,
-        BUILDER.craft_ktlint_task,
-        BUILDER.craft_lint_task,
-        BUILDER.craft_compare_locales_task,
-    ):
-        other_tasks[taskcluster.slugId()] = craft_function()
 
-    return (build_tasks, other_tasks)
+def master_push(queue):
+    task_graph = TaskGraph()
+    task_graph.add_task(BUILDER.craft_detekt_task())
+    task_graph.add_task(BUILDER.craft_ktlint_task())
+    task_graph.add_task(BUILDER.craft_lint_task())
+    task_graph.add_task(BUILDER.craft_compare_locales_task())
 
+    for variant in build_variants.from_gradle():
+        task_graph.add_task(BUILDER.craft_test_task(variant))
+        assemble_task_id = task_graph.add_task(BUILDER.craft_assemble_task(variant))
 
-def nightly(track):
-    is_staging = track == 'staging-nightly'
-    architectures = ['x86', 'arm', 'aarch64']
+        arch, build_type, _ = _get_architecture_and_build_type_and_product_from_variant(variant)
+        # autophone only supports arm and aarch64, so only sign/perftest those builds
+        if build_type == 'debug' and arch in ('arm', 'aarch64'):
```

Comment:
Nit: Use `ReleaseRaptor` instead of debug.

Reply:
Raptor works with `release` now? Awesome :)